lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -510,6 +510,9 @@ ImpellerAllocator::ImpellerAllocator(
 
 std::optional<std::shared_ptr<impeller::DeviceBuffer>>
 ImpellerAllocator::GetDeviceBuffer() const {
+  if (buffer_.has_value()) {
+    buffer_.value()->Flush();
+  }
```

Comment:
Is this the right time to flush?

Why not flush it explicitly in the blit pass where it's getting used?

Reply:
we only need to flush here because we've written directly to the ptr due to the skia interop. Adding the flush anywhere else would cause extra work, as if the buffer was populated with setcontents that would be redundant

Reply:
I misread which class this was in. LGTM.